core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -700,6 +704,11 @@ public TypeProvider getTypes()
             return types;
         }
 
+        public long getPhysicalWrittenDataSize()
```

Comment:
This is done for consistency since we have separate methods for other getters too in `LocalExecutionPlanContext`. For instance, `getDynamicFiltersCollector`.

https://github.com/trinodb/trino/pull/13111#discussion_r927319045

Reply:
Having this indirection will cause the `LocalExecutionPlanContext` object to be retained through the entire lifecycle of the task. It's probably not a big deal, but if it is avoidable I would recommend not to retain it (same for `setWritersFullyScaled`). But yeah, this problem had been already there (`addLocalDynamicFilters`, `getCoordinatorDynamicFilterDomainsCollector`, CC: @raunaqmorarka )

Reply:
Created https://github.com/trinodb/trino/pull/13732 for that